account_bank_statement_import_online_ponto/models/online_bank_statement_provider_ponto.py

```diff
@@ -0,0 +1,244 @@
+                        date = self._ponto_date_from_string(
+                            transaction.get('attributes', {}).get('executionDate'))
+                        if date_since <= date < date_until:
+                            current_transactions.append(transaction)
+
+                    if current_transactions:
+                        if not page_next or (page_next and not latest_identifier):
+                            latest_identifier = current_transactions[0].get('id')
+                        transaction_lines.extend(current_transactions)
+            else:
+                raise UserError(
+                    _('Error during get transaction.\n\n%s \n\n %s') % (
+                        response.status_code, response.text))
+        if latest_identifier:
+            self.ponto_last_identifier = latest_identifier
+        return transaction_lines
+
+    def _ponto_date_from_string(self, date_str):
+        return datetime.strptime(date_str, '%Y-%m-%dT%H:%M:%S.%fZ')
+
+    def _ponto_obtain_statement_data(self, date_since, date_until):
+        self.ensure_one()
+        StatementLine = self.env['account.bank.statement.line']
+        Statement = self.env['account.bank.statement']
+
+        account_ids = self._ponto_get_account_ids()
+        journal = self.journal_id
+
+        iban = self.account_number
+        account_id = account_ids.get(iban)
+        if not account_id:
+            raise UserError(
+                _('Ponto : wrong configuration, unknow account %s')
+                % journal.bank_account_id.acc_number)
+
+        self._ponto_synchronisation(account_id)
+
+        transaction_lines = self._ponto_get_transaction(account_id,
+                                                        date_since,
+                                                        date_until)
+
+        new_transactions = []
+        sequence = 0
+        for transaction in transaction_lines:
+            if StatementLine.sudo().search(
+                [('unique_import_id', '=', transaction['id']),
+                 ('journal_id', '=', journal.id)], limit=1):
+                continue
+
+            sequence += 1
+            attributes = transaction.get('attributes', {})
+            ref = '%s %s' % (
+                attributes.get('description'),
+                attributes.get('counterpartName'))
+            date = self._ponto_date_from_string(attributes.get('executionDate'))
+
+            vals_line = {
+                'sequence': sequence,
+                'date': date,
+                'name': re.sub(' +', ' ', ref) or '/',
+                'ref': attributes.get('remittanceInformation', ''),
+                'unique_import_id': transaction['id'],
+                'amount': attributes['amount'],
+            }
+            new_transactions.append(vals_line)
+
+        if new_transactions:
+            date = max([t['date'] for t in new_transactions])
+            start_balance = Statement.search(
```

Comment:
Please stop faking statement balance if you don't have one!

Reply:
iI understand, if not you should write by hand the balance for each statement.

Reply:
That's what Odoo does with manual statement sand that's why it makes difference between `balance_end` and `balance_end_real` - to require used to double check the data unless it's provided. And faking the data is only misleading users that it's correct and verified.